### ISSUE TITLE
Issue: CORS not working

### ISSUE THREAD

**Issue by wassim** [fruitcake/laravel-cors#432]
**Before you start**
[X] Update to the latest version by running `composer update fruitcake/laravel-cors`

**Check your config**
[X]  Double-check your config file with the version from the repo. Make sure the `paths` property is correctly set. Start by allowing as much as possible.
[X]  Make sure the middleware is added to the global middleware in your Http Kernel (not group)

**Clear your caches**
Please do these steps again before submitting an issue:
[X]  Clear your config cache `php artisan config:clear`, route cache (`php artisan route:clear`) and normal cache (`php artisan cache:clear`).
[X]  Make sure your permissions are setup correctly (eg. storage is writable)

**Kernel.php**

```
<?php

namespace App\Http;

use Illuminate\Foundation\Http\Kernel as HttpKernel;

class Kernel extends HttpKernel
{
    /**
     * The application's global HTTP middleware stack.
     *
     * These middleware are run during every request to your application.
     *
     * @var array
     */
    protected $middleware = [
        \App\Http\Middleware\TrustProxies::class,
        \Fruitcake\Cors\HandleCors::class,
        \App\Http\Middleware\CheckForMaintenanceMode::class,
        \Illuminate\Foundation\Http\Middleware\ValidatePostSize::class,
        \App\Http\Middleware\TrimStrings::class,
        \Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull::class,
    ];

    ...
```

**Cors.php**

```
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Cross-Origin Resource Sharing (CORS) Configuration
    |--------------------------------------------------------------------------
    |
    | Here you may configure your settings for cross-origin resource sharing
    | or "CORS". This determines what cross-origin operations may execute
    | in web browsers. You are free to adjust these settings as needed.
    |
    | To learn more: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
    |
    */

    'paths' => ['api/*'],

    'allowed_methods' => ['*'],

    'allowed_origins' => ['*'],

    'allowed_origins_patterns' => [],

    'allowed_headers' => ['*'],

    'exposed_headers' => false,

    'max_age' => false,

    'supports_credentials' => true,

];
```

**routes/api.php**

```
Route::middleware('auth:api')->group(function () {
    Route::post('/handle', 'Api\MyController@handle');
});
```

**Request headers**

```
Accept: application/json
Authorization: Bearer [hidden_api_token]
Content-Type: application/json;charset=UTF-8
DNT: 1
Referer: https://mywebsite.com/
Sec-Fetch-Dest: empty
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36
```

**Response**

```
Access to XMLHttpRequest at 'https://backend.mywebsite.com/api/handle' from origin 'https://mywebsite.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
index.js:35 Error: Network Error
    at LYNF.e.exports (createError.js:17)
    at XMLHttpRequest.p.onerror (xhr.js:91)
xhr.js:183 POST https://backend.mywebsite.com/api/handle net::ERR_FAILED
```

The request is working fine with a dev tool like Postman.

**Comment by vizko** [comment]
same here, app is offline since I upgraded to laravel 7 😓

**Comment by barryvdh** [comment]
@wassim Did you remove the 'Origin' header or was it not present in the Request? If no Origin header is present, CORS headers will not be added.

**Comment by barryvdh** [comment]
Are you using `fetch()`? Because the default mode is `no-cors`, right?
https://fetch.spec.whatwg.org/#requests
Can you set it to `cors`

**Comment by jamal-rahimzadegan** [comment]
same issue in laravel 7

**Comment by barryvdh** [comment]
Please create your own issue.

**Comment by phambinh217** [comment]
Same

**Comment by NaungYeHtet** [comment]
still open?

**Comment by jamal-rahimzadegan** [comment]
Bo

**Comment by barryvdh** [comment]
I'm closing this due to inactivity. If this issue is still relevant on the latest version (2.0.1 at the moment), please open a new issue with detailed information.
